### PR TITLE
rekor: wait for services to start to avoid crashloop

### PR DESCRIPTION
**Pull request by cpanato**
#### Summary
when installing the rekor chart notice some pods are in crashloop

```
$ k get po
NAME                                         READY   STATUS             RESTARTS   AGE
rekor-mysql-756d7b8b4f-gcn82                 1/1     Running            0          2m11s
rekor-redis-6d5c5c99d8-5t8bn                 1/1     Running            0          2m11s
rekor-server-579c97969-zmffm                 0/1     CrashLoopBackOff   4          2m11s
rekor-trillian-log-server-7b659656bb-9pnp9   0/1     CrashLoopBackOff   4          2m11s
rekor-trillian-log-signer-c8ccc8cfc-8hsrk    0/1     CrashLoopBackOff   3          2m11s
```

this is because `trillian` needs the mysql be up and running to run and `rekor-server` needs `trillian` 
adding some init containers to perform the check before trying to start the services

after the change

```
$ k get po
NAME                                         READY   STATUS     RESTARTS   AGE
rekor-mysql-756d7b8b4f-885f2                 0/1     Running    0          16s
rekor-redis-6d5c5c99d8-9gx6n                 1/1     Running    0          16s
rekor-server-6bc774f9c6-d5mhd                0/1     Init:0/1   0          16s
rekor-trillian-log-server-59b5f5d7b-g52zx    0/1     Init:0/1   0          16s
rekor-trillian-log-signer-579cf4cc9f-v7hdx   0/1     Init:0/1   0          16s


$ k get po
NAME                                         READY   STATUS            RESTARTS   AGE
rekor-mysql-756d7b8b4f-885f2                 1/1     Running           0          56s
rekor-redis-6d5c5c99d8-9gx6n                 1/1     Running           0          56s
rekor-server-6bc774f9c6-d5mhd                0/1     Init:0/1          0          56s
rekor-trillian-log-server-59b5f5d7b-g52zx    0/1     PodInitializing   0          56s
rekor-trillian-log-signer-579cf4cc9f-v7hdx   0/1     PodInitializing   0          56s


$ k get po
NAME                                         READY   STATUS    RESTARTS   AGE
rekor-mysql-756d7b8b4f-885f2                 1/1     Running   0          68s
rekor-redis-6d5c5c99d8-9gx6n                 1/1     Running   0          68s
rekor-server-6bc774f9c6-d5mhd                0/1     Running   0          68s
rekor-trillian-log-server-59b5f5d7b-g52zx    1/1     Running   0          68s
rekor-trillian-log-signer-579cf4cc9f-v7hdx   1/1     Running   0          68s


$ k get po
NAME                                         READY   STATUS    RESTARTS   AGE
rekor-mysql-756d7b8b4f-885f2                 1/1     Running   0          83s
rekor-redis-6d5c5c99d8-9gx6n                 1/1     Running   0          83s
rekor-server-6bc774f9c6-d5mhd                1/1     Running   0          83s
rekor-trillian-log-server-59b5f5d7b-g52zx    1/1     Running   0          83s
rekor-trillian-log-signer-579cf4cc9f-v7hdx   1/1     Running   0          83s
```

#### Ticket Link

Fixes

#### Release Note

```release-note
rekor: wait for services to start to avoid crashloop
```

### CODE REVIEW

**Review by sabre1041 — DISMISSED** [review]
LGTM

File: charts/rekor/templates/server/deployment.yaml [hunk 1]
```diff
@@ -35,8 +35,12 @@ spec:
         {{- end}}
     spec:
       serviceAccountName: {{ template "rekor.serviceAccountName.server" . }}
-      {{- if .Values.server.extraInitContainers }}
       initContainers:
+        - name: "wait-for-trillian-log-server"
+          image: "{{ template "rekor.image" .Values.initContainerImage }}"
+          imagePullPolicy: {{ .Values.initContainerImage.imagePullPolicy }}
+          command: ["sh", "-c", "until curl --max-time 10 http://{{ .Release.Name }}-trillian-log-server:8090; do echo waiting for {{ .Release.Name }}-trillian-log-server; sleep 5; done;"]
```
[the remark this answers is not in the document]
Comment (sabre1041) [reply charts/rekor/templates/server/deployment.yaml:42]: @dol good call out

**Comment by sabre1041** [comment]
@dol good call outs. going to retract my approval for now as your suggestions are well rounded

File: charts/rekor/values.yaml [hunk 2]
```diff
@@ -1,5 +1,12 @@
 imagePullSecrets:
 
+initContainerImage:
+  registry: docker.io
+  repository: appropriate/curl
+  # latest from 2021-10-28
+  version: sha256:c8bf5bbec6397465a247c2bb3e589bb77e4f62ff88a027175ecb2d9e4f12c9d7
```
[the remark this answers is not in the document]
Comment (cpanato) [reply charts/rekor/values.yaml:7]: thanks for all the feedback! i will work on those this week/weekend

**Review by sabre1041 — APPROVED** [review]
LGTM

**Pull request merged**